### PR TITLE
WIP: UI reborn ! :)

### PR DESCRIPTION
**Pull request by Dumbeldor**
_I think i can remove e2e and unit test file to clean._

There are a lot of events to redo

**First event core to ui:**
- [ ] init
- [x] activate
- [x] deactivate
- [x] entergame
- [x] exitgame
- [ ] openingmenu
- [x] systemmessage
- [ ] message
- [x] connect
- [x] disconnect
- [ ] namechange
- [ ] versionset
- [ ] debug
- [ ] debugdata
- [ ] userdataset
- [ ] playerconnected
- [ ] playerdisconnected
- [ ] healthset
- [ ] setlevel
- [ ] setcell
- [ ] setplayer3Dloaded
- [ ] serverid
- [ ] protocolmismatch

**and then event ui to core:**
- [x] connect
- [ ] disconnect
- [ ] sendmessage
- [ ] deactivate

### CODE REVIEW

File: Code/client/include/Services/OverlayService.h [hunk 1]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
     void Create(RenderSystemD3D11* apRenderSystem);
 
-    void Render() const;
+    void Render();
     void Reset() const;
 
+    void Initialize();
+
+    void SetActive(bool aActive);
+    bool GetActive();
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:34]: [[nodiscard]] bool IsActive() const noexcept;

File: Code/client/include/Services/OverlayService.h [hunk 2]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
 
     void Create(RenderSystemD3D11* apRenderSystem);
 
-    void Render() const;
+    void Render();
     void Reset() const;
 
+    void Initialize();
+
+    void SetActive(bool aActive);
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:33]: noexcept

File: Code/client/include/Services/OverlayService.h [hunk 3]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
+public:
+    OverlayService(World& aWorld, entt::dispatcher& aDispatcher);
     ~OverlayService() noexcept;
 
     TP_NOCOPYMOVE(OverlayService);
 
     void Create(RenderSystemD3D11* apRenderSystem);
 
-    void Render() const;
+    void Render();
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:28]: noexcept

File: Code/client/include/Services/OverlayService.h [hunk 4]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
     void Reset() const;
 
+    void Initialize();
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:31]: noexcept

File: Code/client/include/Services/OverlayService.h [hunk 5]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
+
+    void SetActive(bool aActive);
+    bool GetActive();
+
+    void SetInGame(bool aInGame);
+    bool GetInGame();
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:37]: [[nodiscard]] bool IsInGame() const noexcept;

File: Code/client/include/Services/OverlayService.h [hunk 6]
```diff
@@ -9,24 +9,48 @@ namespace TiltedPhoques
...
-    void Render() const;
+    void Render();
     void Reset() const;
 
+    void Initialize();
+
+    void SetActive(bool aActive);
+    bool GetActive();
+
+    void SetInGame(bool aInGame);
```
Comment (yamashi) [inline Code/client/include/Services/OverlayService.h:36]: noexcept

**Review by yamashi — CHANGES_REQUESTED** [review]

File: Code/client/include/Services/OverlayService.h (hunk 1 above)
Comment (Dumbeldor) [reply Code/client/include/Services/OverlayService.h:34]: Thanks, it's nice !

**Comment by Dumbeldor** [comment]
I added checkboxes to follow the evolution of PR. And this PR depends on this one https://github.com/tiltedphoques/TiltedUI/pull/9

**Comment by algorithmhash** [comment]
Gyp error could be resolved with yarn fix by installing with yarn. I can start here and take a look. Thanks @yamashi .

**Comment by Dragonisser** [comment]
I fixed disconnecting and chat on my fork on the feature/updated_ui branch

**Pull request closed without merging**